### PR TITLE
Fix usage of instruction source location

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
get_goto_location should be called on the source_locationt contained in
the instruction, rather than on the one in the code of the instruction,
as that is where they are stored in store_goto_locations.

With this change `goto-instrument --show-goto-code --json-ui` will output the originalGOTOLocation as it's supposed to.

### CODE REVIEW

**Comment by NlightNFotis** [comment]
This is related to the `original GOTO location` work we did sometime in the summer @thk123 . I am not entirely sure if the work was broken from the beginning and this patch is fixing it or if it's working as intended and this patch is breaking it.

This is why I would prefer to see @thk123 's take on this before it gets merged.

**Comment by thk123** [comment]
I'm not sure I understand - I thought there were tests that demonstrated this working before, is that not the case? (In that case, they should be added)

**Comment by hannes-steffenhagen-diffblue** [comment]
@thk123 I can't find any tests for store goto locations with `--json-ui`. Maybe I'm just not seeing them, but if there indeed just aren't any I'll add them.

*EDIT*: I actually can't find any evidence anything json-ui is tested at all.

**Comment by NlightNFotis** [comment]
@thk123 I understand now that this is fixing the json view. When we worked on that, we only worked on the plain text version of that, or at least the changes back then seem to have impacted only that. This is making it work with json more reliably.

**Review by NlightNFotis — APPROVED** [review]

**Review by thk123 — APPROVED** [review]

File: src/goto-programs/show_goto_functions_json.cpp [hunk 1]
```diff
@@ -63,15 +63,16 @@ json_objectt show_goto_functions_jsont::convert(
...
 
-        if(instruction.code.source_location().is_not_nil())
+        if(instruction.source_location.is_not_nil())
         {
           instruction_entry["sourceLocation"]=
-            json(instruction.code.source_location());
+            json(instruction.source_location);
 
-          if(instruction.code.source_location().get_goto_location()!="")
+          if(instruction.source_location.get_goto_location()!="")
```
Comment (peterschrammel) [inline src/goto-programs/show_goto_functions_json.cpp]: Use `!...empty()` ?

**Pull request merged**